### PR TITLE
Added better config support

### PR DESCRIPTION
**Pull request by PatrickShaw**
Needed a way of configuring the server without having to go into the server's config code and changing it. Kept accidentally adding authentication details to the git commit history. This PR should remove the possibility of that happening.

Pretty straightforward I think. You basically just add `openapi-platform.config.json` (or a yaml or json5 file) to wherever you run the server from (for development purposes, probably just `openapi-platform/`) and export a config object.

Will add some alternative ways to configure the server in later PRs.

### CODE REVIEW

File: packages/server/src/config/index.tsx [hunk 1]
```diff
@@ -1,73 +1,77 @@
...
-
-    /**
-     * The password of the PostgreSQL database to connect to.
-     */
-    readonly databasePassword: string;
-  };
+const cwd = process.cwd();
+// TODO: We can provide more ways to specify configs in the future
+// tslint:disable:no-var-requires
+const rawConfig = require(join(cwd, 'openapi-platform.config'));
```
[the remark this answers is not in the document]
Comment (calebj0seph) [reply packages/server/src/config/index.tsx]: I agree that it isn't a likely vulnerability and if someone can modify the config file there are worse things to worry about, but it still makes things easier for an attacker.

[the remark this answers is not in the document]
Comment (calebj0seph) [reply packages/server/src/config/index.tsx]: We also should probably discuss security at some point and come up with a list of things to check/implement.

[the remark this answers is not in the document]
Comment (PatrickShaw) [reply packages/server/src/config/index.tsx]: Yeah i'm down for a war game.

File: packages/server/src/config/index.tsx [hunk 2]
```diff
@@ -1,73 +1,77 @@
...
+      doc: 'The username of the PostgreSQL database to connect to.',
+      env: 'DATABASE_USERNAME',
+      format: String,
+      default: undefined,
+    },
+    password: {
+      doc: 'The password of the PostgreSQL database to connect to.',
+      env: 'DATABASE_PASSWORD',
+      format: String,
+      default: undefined,
```
[the remark this answers is not in the document]
Comment (PatrickShaw) [reply packages/server/src/config/index.tsx]: Done

**Pull request merged**